Update week 10 dg (#118)
AY2223S1-CS2103T-W15-1/tp: docs/DeveloperGuide.md

* `UniqueTagList#editTag(Tag oldTag, Tag newTag)` - Changes the old tag with the new tag.
* `UniquePersonList#changeRelevantPersonTag(oldTag, newTag)` - Updates every person who has the old tag with the new tag.
* `UniqueTodoList#changeRelevantTodoTag(Tag oldTag, Tag newTag)` -  Updates every task which has the old tag with the new tag.

These operations are exposed in the `Model` interface under the same method name.

Given below is an example usage scenario and how the tag editing mechanism behaves at each step.

Step 1. The user executes `tag edit t/friend t/bestFriend` command to edit the old tag, `friend`, to the new tag, `bestFriend`.
`TageditCommandParser` calls  `ArgumentTokenizer#tokenizeToList()` to separate the parameters of `t/friend` and `t/bestFriend`.

Step 2. The `tag edit` command edits the old tag with the new tag, calling `Model#editTag(oldTag, newTag)`.

Step 3. The old tag on every person and every task is now replaced with the new tag for display.

The following activity diagram summarizes what happens when a user executes a tag add command:

(insert activity diagram here)

#### Design consideration

**Aspect: How to implement tag edit:**

> * **Alternative 1 (current choice):** Creates a new tag and replace the old tag with the new one.

Reviewer: ```suggestion
* **Alternative 1 (current choice):** Creates a new tag and replaces the old tag with the new one.
```

Author: Ok. I'll make the changes.

    * Pros: Prevents direct access into the information of a tag.
    * Cons: Tedious. Necessary to manually change the old tag in every person and every task.

> * **Alternative 2:** Change the tag's name .

Reviewer: ```suggestion
* **Alternative 2:** Change the tag's name.
```

Author: Done!

    * Pros: Easy to implement.

>     * Cons: Easy to access into a tag. Could cause accidental bugs.

Reviewer: ```suggestion
    * Cons: Can potentially introduce bugs due to direct access into the tag's implementation.
```

Author: Changes made.
